vespa-cloud/vespa-documentation-search/src/main/java/ai/vespa/cloud/docsearch/OutLinksDocumentProcessor.java

```diff
@@ -64,26 +77,58 @@ public Progress process(Processing processing) {
                 if (document.getDataType().isA(DOC_DOCUMENT_TYPE)) {
                     String myPath = document.getFieldValue(PATH_FIELD_NAME).toString();
 
-                    Set<String> docsLinkedFromThis = canonicalizeLinks(Path.of(myPath),
-                            onlyFileLinks(removeLinkFragment(document)));
+                    Set<String> docsLinkedFromMe = canonicalizeLinks(Path.of(myPath),
+                            onlyFileLinks(removeLinkFragment(getUniqueOutLinks(document))));
 
-                    Array<StringFieldValue> sfv = new Array<>(DataType.getArray(DataType.STRING));
-                    for (String link : docsLinkedFromThis) {
-                        sfv.add(new StringFieldValue(link));
+                    Array<StringFieldValue> sanitizedLinks = new Array<>(DataType.getArray(DataType.STRING));
+                    for (String link : docsLinkedFromMe) {
+                        sanitizedLinks.add(new StringFieldValue(link));
                     }
-                    document.setFieldValue(OUTLINKS_FIELD_NAME, sfv);
 
-                    if (docsLinkedFromThis.size() > 0) {
-                        addInLinks(docsLinkedFromThis, myPath, document.getDataType());
+                    document.setFieldValue(OUTLINKS_FIELD_NAME, sanitizedLinks);
+
+                    if (docsLinkedFromMe.size() > 0) {
+                        addInLinksToOtherDocs(docsLinkedFromMe, myPath, document.getDataType());
                     }
                 }
             }
+            else if (op instanceof DocumentUpdate) {
+                DocumentUpdate update = (DocumentUpdate) op;
+
+                /*
+                  There are two kinds of updates:
+                  * a create-if-nonexistent update for a new document
+                  * updating inlinks to documents
+                 */
+
+                if (update.getFieldUpdate(INLINKS_FIELD_NAME) != null) { return Progress.DONE;}  // no extra processing of inlinks update
```

Comment:
Does this happen as a result of the other processing done in here? That would be worth a mention.

Reply:
added more comments